PublicSDKAPITester/RevenueCatAPI.m

```diff
@@ -0,0 +1,120 @@
+//
+//  RevenueCatAPI.m
+//  MigrateTester
+//
+//  Created by Joshua Liebowitz on 6/18/21.
+//
+
+#import "RevenueCatAPI.h"
+#import "Purchases.h"
+#import "RCPurchases.h"
+
+@import StoreKit;
+
+@implementation RevenueCatAPI
+
+bool canI;
+NSString *version;
+
+BOOL automaticAppleSearchAdsAttributionCollection;
+BOOL debugLogsEnabled;
+RCLogLevel logLevel;
+NSURL *proxyURL;
+BOOL forceUniversalAppStore;
+BOOL simulatesAskToBuyInSandbox;
+RCPurchases *sharedPurchases;
+BOOL isConfigured;
+BOOL allowSharingAppStoreAccount;
+BOOL finishTransactions;
+id<RCPurchasesDelegate> delegate;
+NSString *appUserID;
+BOOL isAnonymous;
+
++ (void)allTheThings {
+//    typedef void (^RCReceivePurchaserInfoBlock)(RCPurchaserInfo * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.ReceivePurchaserInfoBlock);
+//    typedef void (^RCReceiveIntroEligibilityBlock)(NSDictionary<NSString *, RCIntroEligibility *> *) NS_SWIFT_NAME(Purchases.ReceiveIntroEligibilityBlock);
+//    typedef void (^RCReceiveOfferingsBlock)(RCOfferings * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.ReceiveOfferingsBlock);
+//    typedef void (^RCReceiveProductsBlock)(NSArray<SKProduct *> *) NS_SWIFT_NAME(Purchases.ReceiveProductsBlock);
+//    typedef void (^RCPurchaseCompletedBlock)(SKPaymentTransaction * _Nullable, RCPurchaserInfo * _Nullable, NSError * _Nullable, BOOL userCancelled) NS_SWIFT_NAME(Purchases.PurchaseCompletedBlock);
+//    typedef void (^RCDeferredPromotionalPurchaseBlock)(RCPurchaseCompletedBlock);
+//    typedef void (^RCPaymentDiscountBlock)(SKPaymentDiscount * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.PaymentDiscountBlock);
+
+    //    [p presentCodeRedemptionSheet]; ///TODO: iOS ONLY, TEST FOR THIS API BY LOOKING UP SELECTOR
```

Comment:
why not use
```objective-c
#if TARGET_OS_IOS
```
?

Reply:
I spent very little time working around this, but my thoughts are, let's get the APITester up and running, then make it better. With this API, I was hoping I could figure out a way of seeing if it existed during build time without the #if checks. That way we only need 1 target for the APITester. For now, I'd like to leave as a TODO so I can continue thinking about it.